### PR TITLE
Update Newtonsoft.Json from 13.0.1 to 13.0.3

### LINKED ISSUE #12858: Upgrade Newtonsoft.Json reference to 13.0.3
.NET repos currently have a mix of Newtonsoft.Json versions that we're trying to get consolidated into the latest version. This is particularly necessary for the .NET's [source-build](https://github.com/dotnet/source-build) which can only reference one version of a dependency. The latest patch version of Newtonsoft.Json is 13.0.3.

Currently, NuGet.Client is referencing 13.0.1:

https://github.com/NuGet/NuGet.Client/blob/e4a9922edc29c36e6458bc5d3b68a04e0ea14b91/Directory.Packages.props#L6

### PR DESCRIPTION
**Pull request by mthalman**
## Bug

Fixes: https://github.com/NuGet/Home/issues/12858

Regression? Last working version:

## Description
.NET repos currently have a mix of Newtonsoft.Json versions that we're trying to get consolidated into the latest version. This is particularly necessary for the .NET's [source-build](https://github.com/dotnet/source-build) which can only reference one version of a dependency.

Related to https://github.com/dotnet/source-build-externals/pull/195

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Comment by mthalman** [comment]
cc @MichaelSimons for review

**Review by MichaelSimons — APPROVED** [review]

File: test/EndToEnd/tests/BuildIntegratedTest.ps1 [hunk 1]
```diff
@@ -216,7 +216,7 @@ function Test-BuildIntegratedProjectClosureWithLegacyProjects {
 function Test-BuildIntegratedMixedLegacyProjects {
     # Arrange
     $project1 = New-ClassLibrary
-    $project1 | Install-Package Newtonsoft.Json -Version 13.0.1
```
Comment (nkolev92) [inline test/EndToEnd/tests/BuildIntegratedTest.ps1]: This is test code. Please revert this.

**Review by nkolev92 — CHANGES_REQUESTED** [review]
Hey @mthalman 

Thanks for the PR. 
Please follow the PR template. We depend on that for automation. For product changes like this, we need a NuGet/Home tracking issue as per the template. 

I don't think we need to change the test code either. Those tests are not run in source build scenarios anyways.

**Review by nkolev92 — APPROVED** [review]
Change looks good. Our insertions are blocked today, so not sure if this would stand a chance at making P2 in VS. 

The SDK seems to be using 13.0.1. https://github.com/dotnet/sdk/blob/main/eng/Versions.props#L31. 
Do we need to coordinate with them?

**Comment by mthalman** [comment]
> The SDK seems to be using 13.0.1. https://github.com/dotnet/sdk/blob/main/eng/Versions.props#L31.

Yes, I've got a PR for the SDK changes: https://github.com/dotnet/sdk/pull/35008

Back-references for all PRs related to this work can be found at https://github.com/dotnet/source-build-externals/pull/195.

**Review by heng-liu — APPROVED** [review]

**Review by zivkan — APPROVED** [review]

**Pull request merged**